src/NodeParser/ReturnTypeNodeParser.ts

```diff
@@ -0,0 +1,134 @@
+            }
+
+            const typeArg = node.typeArguments[0];
+            
+            // Handle different types of type arguments
+            if (ts.isTypeQueryNode(typeArg)) {
+                // Case: ReturnType<typeof functionName>
+                // Get the symbol for the identifier
+                const symbol = this.checker.getSymbolAtLocation(typeArg.exprName);
+                if (!symbol) {
+                    throw new UnknownNodeError(node);
+                }
+
+                // Get the declarations of the symbol
+                const declarations = symbol.getDeclarations() || [];
+
+                // Try multiple methods to extract return type
+                for (const decl of declarations) {
+                    let returnTypeNode: ts.TypeNode | undefined;
+
+                    // If declaration is a function/method with explicit return type
+                    if (
+                        (ts.isFunctionDeclaration(decl) || ts.isMethodDeclaration(decl) || 
+                         ts.isArrowFunction(decl) || ts.isFunctionExpression(decl)) && 
+                        decl.type
+                    ) {
+                        returnTypeNode = decl.type;
+                    } 
+                    // If declaration is a variable with function type annotation
+                    else if (
+                        ts.isVariableDeclaration(decl) && 
+                        decl.type && 
+                        ts.isFunctionTypeNode(decl.type)
+                    ) {
+                        returnTypeNode = decl.type.type;
+                    }
+
+                    // If we found a return type node, process it
+                    if (returnTypeNode) {
+                        const baseType = this.childNodeParser.createType(returnTypeNode, context);
+                        return baseType;
+                    }
+                }
+
+                // Fallback to type checking method
+                const type = this.checker.getTypeOfSymbolAtLocation(symbol, typeArg);
+                const signatures = type.getCallSignatures();
+
+                if (signatures.length > 0) {
+                    // Use getReturnType directly from the signature
+                    const returnType = signatures[0].getReturnType();
+                    
+                    const returnTypeNode = this.checker.typeToTypeNode(
+                        returnType, 
+                        undefined, 
+                        ts.NodeBuilderFlags.NoTruncation
+                    );
+
+                    if (returnTypeNode) {
+                        return this.childNodeParser.createType(returnTypeNode, context);
+                    }
+                }
+            } else {
+                // Case: ReturnType<SomeType["methodName"]> or other complex types
+                // Get the type directly from TypeScript's type system
+                const argType = this.checker.getTypeAtLocation(typeArg);
```

Comment:
I really want to a lid the type checker if possible. What cases fail when we remove this branch?

Reply:
Yeah, but what line of the example causes this?

Reply:
test/valid-data/type-return-type-complex/main.ts(15,28)

https://github.com/vega/ts-json-schema-generator/pull/2317/files#diff-410929871b91b13bf666ba5f7b9b865ef50c52318faaa956ef8e99a6cc996c5aR15 

Specifically the node ReturnType 

If you give me some more context of what to look at, I can take a deeper look to finding a better solution.